pkg/assets/embedded.go

```diff
@@ -0,0 +1,59 @@
+package assets
+
+import (
+	"context"
+	"embed"
+	"fmt"
+	"io/fs"
+	"net/http"
+	"strconv"
+	"strings"
+
+	"github.com/tilt-dev/tilt/pkg/logger"
+)
+
+//go:embed build
+var build embed.FS
+
+type embeddedServer struct {
+	http.Handler
+}
+
+func NewEmbeddedServer() (Server, error) {
```

Comment:
i think the more idiomatic API is something like:

GetEmbeddedServer() (Server, bool)

since "the server doesn't exist" isn't something that you're treating as an error. here's an example of an API that does this: https://pkg.go.dev/os#LookupEnv